main.go

```diff
@@ -143,18 +173,93 @@ func getLocator(ctx context.Context) (*ipfsgeoip.IPLocator, error) {
 
 	go lite.Bootstrap(ipfslite.DefaultBootstrapPeers())
 
-	// An exchange session will speed things up
 	return ipfsgeoip.NewIPLocator(lite.Session(ctx)), nil
 }
 
-func getMinerInfos(db *pg.DB) ([]minerInfo, error) {
+func getRouter(ctx context.Context) (*filecoinPeer, error) {
+	fp, err := setupFilecoinPeer(
+		ctx,
+		"mainnet",
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return fp, nil
+}
+
+func getActiveMiners(db *pg.DB, fp *filecoinPeer) (<-chan minerInfo, error) {
+	go fp.bootstrap()
```

Comment:
I believe this function already does the connecting in the background and returns, so we shouldn't need to launch it as goroutine. It might be desirable to wait for it to finish connecting to bootstrappers before making dht queries though.